semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/vectorsearch/filtering/BasicVectorSearchFilter.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.data.vectorsearch.filtering;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class BasicVectorSearchFilter {
+
+    /**
+     * Creates a new instance of the BasicVectorSearchFilter class.
+     *
+     * @return A new instance of the BasicVectorSearchFilter class.
+     */
+    public static BasicVectorSearchFilter createDefault() {
+        return new BasicVectorSearchFilter();
+    }
+
+    private final List<FilterClause> filterClauses;
+
+    public BasicVectorSearchFilter() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Creates a new instance of the BasicVectorSearchFilter class.
+     *
+     * @param filterClauses The filter clauses.
+     */
+    public BasicVectorSearchFilter(List<FilterClause> filterClauses) {
+        this.filterClauses = new ArrayList<>(filterClauses);
```

Comment:
Why a copy? Also, and this is just a nit, if I use the default constructor I'm creating an array that's just going to get thrown away. If you're never going to modify filterClauses, go ahead and make it an unmodifiableList right here instead of creating a new unmodifiableList in getFilterClauses().